pkg/build/registry/buildconfig/webhook.go

```diff
@@ -151,6 +168,37 @@ func (w *WebHookHandler) ProcessWebHook(writer http.ResponseWriter, req *http.Re
 	return warning
 }
 
+// CheckSecret tests the user provided secret against the secrets for the webhook triggers, if a match is found
+// then the corresponding webhook trigger is returned.
+func CheckSecret(namespace, userSecret string, triggers []*buildapi.WebHookTrigger, secretsClient kcoreclient.SecretsGetter) (*buildapi.WebHookTrigger, error) {
+	for i := range triggers {
+		secretRef := triggers[i].SecretRef
+		secret := triggers[i].Secret
+		if len(secret) > 0 {
+			if hmac.Equal([]byte(secret), []byte(userSecret)) {
+				return triggers[i], nil
+			}
+		}
+		if secretRef != nil {
+			s, err := secretsClient.Secrets(namespace).Get(secretRef.Name, metav1.GetOptions{})
+			if err != nil && !errors.IsNotFound(err) {
+				return nil, err
+			}
+			for _, v := range s.Data {
+				decoded, err := base64.StdEncoding.DecodeString(string(v))
```

Comment:
This is unexpected.  You're storing base64 encoded bytes instead of the actual bytes?

Reply:
you might be looking at this PR by commit?  this is changed in a later commit in the PR (though i see there's a log.info i inadvertently left in)